### PR TITLE
Adding Coding in the Cabana

### LINKED ISSUE #1790: Coding in the Cabana website suggestions
I'm creating this issue as a place to keep suggestions for how to put the new ```coding in the cabana``` series on the website. Currently, there is no way to submit contributions or view the code.

here are some of the current suggestions, comment yours

- use the same coding challenge page but add some visual distinction
- add another page for coding in the cabana videos

### PR DESCRIPTION
**Pull request by violetcraze**
fixes #1790 

As discussed in the last live stream and @GypsyDangerous created issue #1790, I put together a possible new page for Coding in the Cabana videos. Chose to go this route to avoid numbering problems with Coding Challenges. But the new pages follow largely the same format.

I entered Cabana as the short name and it made a `To the Cabana` button on the videos page. I liked it so that's what I put in the nav. 

If you have any suggestions around improvements or naming of things that would be good to discuss now as it'll be harder to change them later on.

Here are some screen shots of what the new pages would look like:
Screenshot from 2019-10-18 16-22-25
Screenshot from 2019-10-18 16-22-28
Screenshot from 2019-10-18 16-22-38

### CODE REVIEW

File: _CodingInTheCabana/001-maurer-rose.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: "Maurer Rose"
+video_number: 1
+date: 2019-10-11
+video_id: 4uU9lZ-HSqA
+repository: Cabana_001_MaurerRose
```
Comment (meiamsome) [inline _CodingInTheCabana/001-maurer-rose.md]: This looks like it is here accidentally - it's being overridden by line 8

Comment (violetcraze) [reply _CodingInTheCabana/001-maurer-rose.md]: Thanks for pointing this out. I missed that it was already in there when I went to add the code. Should be fixed now.

**Comment by meiamsome** [comment]
This all looks good to me, great stuff!

One small thing that would be helpful - could you add the folder to this listing here: https://github.com/CodingTrain/website/blob/master/unit_testing/yaml.test.js#L38-L44

This will make sure our CI scripts that verify the YAML contents for us will go over this and future Coding in the Cabana YAML files.

It should be something like:
```
let directories = {
  _CodingChallenges: formatDefinitions.video,
  _CodingInTheCabana: formatDefinitions.video,
  _Courses: formatDefinitions.video,
  _GuestTutorials: formatDefinitions.video,
  _Streams: formatDefinitions.stream,
  _Tutorials: formatDefinitions.video,
};
```

Hopefully it won't pick up any problems with your YAML, it looks fine to me at least, but computers are often better than humans on these things!

**Comment by violetcraze** [comment]
@meiamsome Do you know if I also have to include it in `yamlFolders` within files.js? https://github.com/CodingTrain/website/blob/0020a8ef070ab305b8b768fcb98e5eac5c68e0eb/unit_testing/files.js#L5

**Comment by meiamsome** [comment]
@violetcraze yep you will, good catch - those should potentially be merged to a single source of truth, but for this PR just leave add it there as well I think!

**Comment by violetcraze** [comment]
@meiamsome Thanks! The tests caught a couple small problems with links. For future reference is there a way to run them locally?

**Comment by meiamsome** [comment]
Yes, you can run them locally by running `npm test` in the `unit_testing` directory - as long as you have node set up and have installed the requirements by running `npm ci` in that directory too. That will get you most of the checking of the CI.


The CI setup also runs some further tests on the output html pages, that you can also run locally if you follow this guide to get github pages working locally: https://help.github.com/en/articles/testing-your-github-pages-site-locally-with-jekyll
And then after building the pages, you should be able to run the check command specified here: https://github.com/CodingTrain/website/blob/master/.circleci/config.yml#L74
These tests are a lot harder to set up to run locally correctly and don't provide too much value, so it's probably not worth setting them up in my opinion.

**Pull request merged**